etc/kayobe/ansible/swap.yml

```diff
@@ -0,0 +1,35 @@
+---
+# Custom playbook to configure a swap device. This may be used as a
+# post-overcloud host configure hook.
+#
+# Variables:
+# * swap_group: Host pattern against which to target the playbook. Default is
+#               'overcloud'.
+# * swap_device: Name of the swap device to configure. Default is
+#                '/dev/rootvg/lv_swap'.
+
+- name: Configure swap
+  hosts: "{{ swap_group | default('overcloud') }}"
+  tags:
+    - swap
+  vars:
+    swap_device: "/dev/rootvg/lv_swap"
+  become: true
+  tasks:
+    - name: Ensure swap filesystem is present
+      filesystem:
+        fstype: "swap"
+        dev: "{{ swap_device }}"
+
+    - name: Ensure swap device present in fstab
+      mount:
+        name: "none"
+        src: "{{ swap_device }}"
+        fstype: "swap"
+        state: "present"
+
+    # It does no harm to run this when swap is already active
```

Comment:
Why use the:
```
      when:
        - ansible_facts.swaptotal_mb == 0
```
condition then?

Reply:
Because that was already in the playbook when I picked it up :)

I guess it also helps with not showing things as changed. There may be cases where we avoid activating swap though, e.g. if facts are stale, or there is another swap device/file.

Should we remove it?

Reply:
Ahh, yeah, of course... For idempotency since that task will always report a changed status. Makes sense, thanks. I did find that in verbose mode swapon produces some output you could parse:
```
[will@juno ~]$ sudo swapon -av
swapon: /swapfile: found signature [pagesize=4096, signature=swap]
swapon: /swapfile: pagesize=4096, swapsize=2147483648, devsize=2147483648
swapon /swapfile
[will@juno ~]$ sudo swapon -av
swapon: /swapfile: already active -- ignored
```
to set changed_when, but feels like bike shedding.